commonmark-ext-gfm-tables/src/test/java/org/commonmark/ext/gfm/tables/TextContentTablesTest.java

```diff
@@ -0,0 +1,179 @@
+package org.commonmark.ext.gfm.tables;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import org.commonmark.Extension;
+import org.commonmark.node.Node;
+import org.commonmark.parser.Parser;
+import org.commonmark.renderer.html.AttributeProvider;
+import org.commonmark.renderer.html.AttributeProviderContext;
+import org.commonmark.renderer.html.AttributeProviderFactory;
+import org.commonmark.renderer.html.HtmlRenderer;
+import org.commonmark.renderer.text.TextContentNodeRendererFactory;
+import org.commonmark.renderer.text.TextContentRenderer;
+import org.commonmark.test.RenderingTestCase;
+import org.junit.Test;
+
+public class TextContentTablesTest extends RenderingTestCase {
+
+    private static final Set<Extension> EXTENSIONS = Collections.singleton(TextContentTablesExtension.create());
+    private static final Parser PARSER = Parser.builder().extensions(EXTENSIONS).build();
+    private static final TextContentRenderer RENDERER = TextContentRenderer.builder().extensions(EXTENSIONS).build();
+
+    @Test
+    public void mustHaveHeaderAndSeparator() {
+        assertRendering("Abc|Def", "Abc|Def");
+        assertRendering("Abc | Def", "Abc | Def");
+    }
+
+    @Test
+    public void separatorMustBeThreeOrMore() {
+        assertRendering("Abc|Def\n-|-", "Abc|Def\n-|-");
+        assertRendering("Abc|Def\n--|--", "Abc|Def\n--|--");
+    }
+
+    @Test
+    public void separatorCanNotHaveLeadingSpaceThenPipe() {
+        assertRendering("Abc|Def\n |---|---", "Abc|Def\n|---|---");
+    }
+
+    @Test
+    public void headerMustBeOneLine() {
+        assertRendering("No\nAbc|Def\n---|---", "No\nAbc|Def\n---|---");
+    }
```

Comment:
This test and the ones above are not necessary, they test parsing instead of rendering. You can remove them from here. This class should only be about the text rendering.